guide/deployment/deploying-to-docker.md

```diff
@@ -0,0 +1,154 @@
+# Deploying to Docker
+
+Running projects like your bot with Docker is a great way to get your bot running on a VPS, with the added benefits of being able to quickly spin up your bot on other servers using Docker, use environment variables, simple scaling, auto updating on GitHub pushes, and much more.
+
+1. Get access to your VPS's terminal via SSH or other methods.
+2. Install Docker on your VPS by following the specific instructions for your linux [distribution](https://docs.docker.com/engine/install/#server)
+3. Create a [Dockerfile](#creating-a-dockerfile) or [docker-compose](#creating-a-docker-compose-file) to run your bot.
+4. [Build](#building-your-docker-image) your Docker image.
+5. [Run](#running-your-docker-container) your bot.
+
+## Creating a Dockerfile
+
+A Dockerfile is a file that tells Docker how to build your bot.
+
+:::tip
+If you have been following the guide so far the below Dockerfile should work for you.
+:::
+
+```dockerfile
+# Use node:16.11.1 as the base image
+FROM node:16.11.1
+
+# Create app directory
+WORKDIR /usr/src/app
+
+# Copy package.json
+COPY package*.json ./
+
+# Fetch dependencies
+RUN npm i
+
+# Copy code
+COPY . .
+
+# Install pm2
+RUN npm install pm2 -g
+
+# Start the bot using pm2 so errors won't kill the container, Learn More: https://discordjs.guide/improving-dev-environment/pm2.html#installation
+CMD [ "pm2-runtime", "start", "index.js" ]
+```
+
+# Building your Docker image
+
+<Definition content="A Docker image is a read-only template containing instructions for creating a container that can run on the Docker platform. It provides an easy way to package up applications and pre-configured server environments that you can use privately or publicly with other Docker users." />
+
+Now that you have your `Dockerfile` and `.dockerignore` file, you can build your Docker image, with your instruction manual you made for the cli. Most people like to [automate this](#automating-the-build-process-with-github-actions) but you can also do it [manually](#manually-building-and-publishing-your-docker-image).
+
+## Manually building and publishing your Docker image
+
+Step 1: Build your Docker image
+
+```bash
+    docker build -t [your-image-name] .
+```
+
+Step 2: Push your Docker image to Docker Hub
```

Comment:
Doesn't this step combined with `COPY . .` in your Dockerfile cause the user's `config.json`  (which includes their bot token) to be uploaded to the cloud? This has the potential to be dangerous, especially if you don't warn the user.

Reply:
https://github.com/discordjs/guide/pull/1047/commits/8479aff9ac2e1ad7cbb321c178557facc7111241#diff-42e8c3f5c3b69c37706bd00901b745166e411832a887b85b2b09a37fb73fed12R5